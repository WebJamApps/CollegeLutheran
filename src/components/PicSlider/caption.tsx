import React from 'react';

const Caption = (props: { caption?: (JSX.Element|string); }):JSX.Element => {
  const { caption } = props;
  return (
    <div
      className="slider-caption"
      style={{
        textAlign: 'center',
        fontWeight: 'bold',
        padding: 0,
        backgroundColor: 'silver',
        marginBottom: 0,
        marginTop: 0,
      }}
    >
      {' '}
      {caption}
      {' '}
    </div>
  );
};

Caption.defaultProps = {
  caption: '',
};

export default Caption;
