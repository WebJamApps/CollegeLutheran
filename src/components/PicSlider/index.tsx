import React, { Component } from 'react';
import Slider from 'react-slick';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import Caption from './caption';
import { Ibook } from '../../redux/mapStoreToProps';

type SettingsType = {
  autoplay: boolean;
  autoplaySpeed: number;
  infinite: boolean;
  speed: number;
  slidesToShow: number;
  slidesToScroll: number;
  arrows: boolean;
  fade: boolean;
};

export interface PicSliderProps {
  settings?: SettingsType;
  data: Ibook[];
}
class PicSlider extends Component<PicSliderProps> {
  static defaultProps: { data: [{ url: ''; title: ''; _id: 0 }]; };

  settings: SettingsType;

  constructor(props: Readonly<PicSliderProps>) {
    super(props);
    this.settings = {
      autoplay: true,
      autoplaySpeed: 3000,
      infinite: true,
      speed: 5000,
      slidesToShow: 1,
      slidesToScroll: 1,
      arrows: false,
      fade: true,
    };
  }

  /* eslint-disable react/jsx-props-no-spreading */
  render(): JSX.Element {
    const { data } = this.props;
    return (
      <div>
        <Slider {...this.settings}>
          {
            data.map((d: Ibook) => (
              <div key={d._id}>
                <img className="slide-show" src={d.url} alt={d.title} />
                {d.comments === 'showCaption' ? <Caption caption={d.title} /> : null}
              </div>
            ))
          }
        </Slider>
      </div>
    );
  }
}

export default PicSlider;
