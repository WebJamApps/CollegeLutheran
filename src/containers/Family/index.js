import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DefaultFamilyContent from './FamilyContent';
import mapStoreToProps from '../../redux/mapStoreToProps';
import commonUtils from '../../lib/commonUtils';

export const Family = ({ familyPics }) => {
  commonUtils.setTitleAndScroll('Children & Families');
  return (<DefaultFamilyContent familyPics={familyPics} />);
};

Family.defaultProps = { familyPics: [] };
Family.propTypes = {
  familyPics: PropTypes.arrayOf(PropTypes.shape({
    comments: PropTypes.string,
  })),
};

export default connect(mapStoreToProps, null)(Family);
