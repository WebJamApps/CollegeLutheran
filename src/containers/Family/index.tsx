import React from 'react';
import { connect } from 'react-redux';
import DefaultFamilyContent from './FamilyContent';
import mapStoreToProps, { Ibook } from '../../redux/mapStoreToProps';
import commonUtils from '../../lib/commonUtils';

export interface FamilyProps {
  familyPics?: Ibook[];
}

export const Family = ({ familyPics }: FamilyProps): JSX.Element => {
  commonUtils.setTitleAndScroll('Children & Families', window.screen.width);
  return (<DefaultFamilyContent familyPics={familyPics} />);
};

Family.defaultProps = { familyPics: [] };

export default connect(mapStoreToProps, null)(Family);
