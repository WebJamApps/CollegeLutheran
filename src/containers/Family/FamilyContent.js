import React from 'react';
import PropTypes from 'prop-types';
import ReactHtmlParser from 'react-html-parser';
import PicSlider from '../../components/pic-slider';

const FamilyContent = ({ familyPics }) => (
  <div className="page-content">
    <div id="familySlideshowWide" style={{ display: 'flex', flexDirection: 'column' }}>
      {familyPics.length > 0 ? (<PicSlider data={familyPics} />) : null}
      <p style={{
        display: 'flex', alignSelf: 'center', fontSize: '14px', textAlign: 'center',
      }}
      >
        A test to make sure data is being rendered correctly. Still needs to be passed to pic-slider to render all.
      </p>
    </div>
    <div className="container-fluid">
      <p style={{ fontSize: '4pt', margin: '0' }}>&nbsp;</p>
      <div className="material-content elevation3" style={{ maxWidth: '998px', paddingBottom: '-80px', margin: 'auto' }}>
        <h3 style={{ paddingBottom: '15px', fontWeight: '100' }}>Children and Families</h3>
        <p>
          <span style={{ color: 'rgb(0, 51, 102)' }}>
            <strong>
              <i>&quot;The members of the body that seem to be weaker are indispensable.&quot;</i>
              &nbsp;&nbsp;&nbsp;1 Corinthians 12:22
            </strong>
          </span>
        </p>
        <p>
            Children are indispensable at College Lutheran Church. The Children and Family Ministry at College Lutheran Church supports kids in
            preschool through 5th grade and their families. Age appropriate Sunday School, monthly meet-ups, and
            seasonal events such as Trunk or Treating, Christmas Programs, Easter Parties, and Vacation Bible School are a few of the ways that
            we familiarize our children with Biblical stories and the power of living in Christian community.
        </p>
        <p>
            Children are also encouraged to be full participants in worship. They may follow along in the bulletin with a parent or use the
            children’s activity bags located outside the doors at the back of the sanctuary. Both the children’s
            sermon
            and the kid’s food offering are special parts of the service designated for kid participation. Additionally, there
            is a Nursery available during worship (located on the 2nd floor), if parents prefer to have their children enjoy a safe
            play zone.
        </p>
        <p>
            We celebrate the milestones of Baptism and First Communion and offer guidance for parents during these momentous events. If you
            would like your child to be baptized or receive First communion, please contact Pastor David at
          {' '}
          <a href="mailto:pastordrebes@collegelutheran.org">pastordrebes@collegelutheran.org</a>
          <br />
          We invite you and your family to come experience a fun, embracing Christian community at any or all of our upcoming events.
        </p>
        <section style={{ textAlign: 'left', marginTop: '0' }}>
          {' '}
          {ReactHtmlParser(familyPics.comments)}
        </section>
      </div>
      <div style={{
        textAlign: 'center', margin: 'auto', paddingTop: '0', paddingBottom: '-40px',
      }}
      >
        <a href="http://www.elca.org/" target="_blank" rel="noopener noreferrer">
          <img
            id="elcaLogo"
            alt="ELCA LOGO"
            src="https://dl.dropboxusercontent.com/s/wkzubcmmm3pqst4/elca-logo.png?dl=0"
            style={{ width: '350px', paddingTop: '30px', margin: 'auto' }}
          />
        </a>
      </div>
    </div>
  </div>
);
FamilyContent.defaultProps = { familyPics: [] };
FamilyContent.propTypes = {
  familyPics: PropTypes.arrayOf(PropTypes.shape({
    comments: PropTypes.string,
  })),
};
export default FamilyContent;
