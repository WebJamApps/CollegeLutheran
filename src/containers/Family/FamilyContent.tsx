import React from 'react';
import ReactHtmlParser from 'react-html-parser';
import PicSlider from '../../components/PicSlider';
import ELCALogo from '../../components/elcaLogo';
import type { FamilyProps } from './index';

const pageName = 'family';// eslint-disable-next-line max-len
const additionalContent = '<hr><h5>Resources for Families</h5><p><strong><em>Roots and Wings</em></strong>, a monthly publication of the Virginia Synod that provides suggestions of daily family time activities:<a target="_blank" href="https://www.vasynod.org/ministries/roots-and-wings/">https://www.vasynod.org/ministries/roots-and-wings/</a></p><p><strong>Ideas to bring worship to the home:</strong>&nbsp;<a target="_blank" href="https://www.vasynod.org/wp-content/uploads/2012/11/Worship-in-the-Home.pdf">https://www.vasynod.org/wp-content/uploads/2012/11/Worship-in-the-Home.pdf</a></p><p><strong>Hand Prayer:</strong>&nbsp;<a target="_blank" href="http://www.vibrantfaithathome.org/item/hand-prayers">http://www.vibrantfaithathome.org/item/hand-prayers</a></p>';
const FamilyContent = ({ familyPics }: FamilyProps): JSX.Element => (
  <div className="page-content">
    <div id="familySlideshowWide" style={{ display: 'flex', flexDirection: 'column' }}>
      {familyPics && familyPics.length > 0 ? (<PicSlider data={familyPics} />) : null}
    </div>
    <div className="container-fluid">
      <p style={{ fontSize: '4pt', margin: '0' }}>&nbsp;</p>
      <div className="material-content elevation3" style={{ maxWidth: '998px', paddingBottom: '-80px', margin: 'auto' }}>
        <h3 style={{ paddingBottom: '15px' }}>Children and Families</h3>
        <p>
          <span style={{ color: 'rgb(0, 51, 102)' }}>
            <strong>
              <i>&quot;The members of the body that seem to be weaker are indispensable.&quot;</i>
              &nbsp;&nbsp;&nbsp;1 Corinthians 12:22
            </strong>
          </span>
        </p>
        <p>
          Children are indispensable at College Lutheran Church. The Children and Family Ministry at College Lutheran Church supports kids in
          preschool through 5th grade and their families. Age appropriate Sunday School, monthly meet-ups, and
          seasonal events such as Trunk or Treating, Christmas Programs, Easter Parties, and Vacation Bible School are a few of the ways that
          we familiarize our children with Biblical stories and the power of living in Christian community.
        </p>
        <p>
          Children are also encouraged to be full participants in worship. They may follow along in the bulletin with a parent or use the
          children’s activity bags located outside the doors at the back of the sanctuary. Both the children’s
          sermon
          and the kid’s food offering are special parts of the service designated for kid participation. Additionally, there
          is a Nursery available during worship (located on the 2nd floor), if parents prefer to have their children enjoy a safe
          play zone.
        </p>
        <p>
          We celebrate the milestones of Baptism and First Communion and offer guidance for parents during these momentous events. If you
          would like your child to be baptized or receive First communion, please contact Pastor David at
          {' '}
          <a href="mailto:pastordrebes@collegelutheran.org">pastordrebes@collegelutheran.org</a>
          <br />
          We invite you and your family to come experience a fun, embracing Christian community at any or all of our upcoming events.
        </p>
        <section style={{ textAlign: 'left', marginTop: '0' }}>
          {' '}
          {ReactHtmlParser(additionalContent)}
        </section>
      </div>
      <ELCALogo pageName={pageName} />
    </div>
  </div>
);
FamilyContent.defaultProps = { familyPics: [] };

export default FamilyContent;
