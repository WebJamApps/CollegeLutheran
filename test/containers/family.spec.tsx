/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import renderer from 'react-test-renderer';
import { Family } from '../../src/containers/Family/index';
import DefaultFamilyContent from '../../src/containers/Family/FamilyContent';
import PicSlider from '../../src/components/PicSlider';

function setup(data: any[]) {
  let wrapper: ShallowWrapper<any, Readonly<any>, React.Component<any, any, any>>;
  if (data !== null && data !== undefined) {
    wrapper = shallow(<Family familyPics={data} />);
  } else wrapper = shallow(<Family />);
  return { wrapper };
}

describe('Family', () => {
  it('renders correctly without images', () => {
    const tree = renderer
      .create(<Family familyPics={[]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Renders the Family component', () => {
    const data = [{}];
    const { wrapper } = setup(data);
    expect(wrapper.find(DefaultFamilyContent).exists()).toBe(true);
    expect(wrapper.find(DefaultFamilyContent).dive().find('div.page-content').exists()).toBe(true);
  });
  it('renders with images', () => new Promise((done) => {
    const data:any[] = [{
      title: '', type: '', _id: '', created_at: '',
    }];
    const wrapper2 = shallow(<DefaultFamilyContent familyPics={data} />);
    expect(wrapper2.find(PicSlider).exists()).toBe(true);
    done();
  }));
});
