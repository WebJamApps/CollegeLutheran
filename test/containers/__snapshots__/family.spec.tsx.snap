// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Family renders correctly without images 1`] = `
<div
  className="page-content"
>
  <div
    id="familySlideshowWide"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  />
  <div
    className="container-fluid"
  >
    <p
      style={
        Object {
          "fontSize": "4pt",
          "margin": "0",
        }
      }
    >
       
    </p>
    <div
      className="material-content elevation3"
      style={
        Object {
          "margin": "auto",
          "maxWidth": "998px",
          "paddingBottom": "-80px",
        }
      }
    >
      <h3
        style={
          Object {
            "paddingBottom": "15px",
          }
        }
      >
        Children and Families
      </h3>
      <p>
        <span
          style={
            Object {
              "color": "rgb(0, 51, 102)",
            }
          }
        >
          <strong>
            <i>
              "The members of the body that seem to be weaker are indispensable."
            </i>
               1 Corinthians 12:22
          </strong>
        </span>
      </p>
      <p>
        Children are indispensable at College Lutheran Church. The Children and Family Ministry at College Lutheran Church supports kids in preschool through 5th grade and their families. Age appropriate Sunday School, monthly meet-ups, and seasonal events such as Trunk or Treating, Christmas Programs, Easter Parties, and Vacation Bible School are a few of the ways that we familiarize our children with Biblical stories and the power of living in Christian community.
      </p>
      <p>
        Children are also encouraged to be full participants in worship. They may follow along in the bulletin with a parent or use the children’s activity bags located outside the doors at the back of the sanctuary. Both the children’s sermon and the kid’s food offering are special parts of the service designated for kid participation. Additionally, there is a Nursery available during worship (located on the 2nd floor), if parents prefer to have their children enjoy a safe play zone.
      </p>
      <p>
        We celebrate the milestones of Baptism and First Communion and offer guidance for parents during these momentous events. If you would like your child to be baptized or receive First communion, please contact Pastor David at
         
        <a
          href="mailto:pastordrebes@collegelutheran.org"
        >
          pastordrebes@collegelutheran.org
        </a>
        <br />
        We invite you and your family to come experience a fun, embracing Christian community at any or all of our upcoming events.
      </p>
      <section
        style={
          Object {
            "marginTop": "0",
            "textAlign": "left",
          }
        }
      >
         
        <hr />
        <h5>
          Resources for Families
        </h5>
        <p>
          <strong>
            <em>
              Roots and Wings
            </em>
          </strong>
          , a monthly publication of the Virginia Synod that provides suggestions of daily family time activities:
          <a
            href="https://www.vasynod.org/ministries/roots-and-wings/"
            target="_blank"
          >
            https://www.vasynod.org/ministries/roots-and-wings/
          </a>
        </p>
        <p>
          <strong>
            Ideas to bring worship to the home:
          </strong>
           
          <a
            href="https://www.vasynod.org/wp-content/uploads/2012/11/Worship-in-the-Home.pdf"
            target="_blank"
          >
            https://www.vasynod.org/wp-content/uploads/2012/11/Worship-in-the-Home.pdf
          </a>
        </p>
        <p>
          <strong>
            Hand Prayer:
          </strong>
           
          <a
            href="http://www.vibrantfaithathome.org/item/hand-prayers"
            target="_blank"
          >
            http://www.vibrantfaithathome.org/item/hand-prayers
          </a>
        </p>
      </section>
    </div>
    <div
      className="familyELCA"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <a
        href="http://www.elca.org/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="ELCA LOGO"
          id="elcaLogo"
          src="https://dl.dropboxusercontent.com/s/wkzubcmmm3pqst4/elca-logo.png?dl=0"
          style={
            Object {
              "margin": "auto",
              "marginLeft": "-12px",
              "paddingTop": "30px",
              "width": "310px",
            }
          }
        />
      </a>
    </div>
  </div>
</div>
`;
